[Elizabeth Hong] iP

elizabethhky opened the pull request:
# Welcome to UwuBot ٩(๑❛ᴗ❛๑)۶ ♥️ 
> "The only difference between success and failure is the ability to take action." – Alexandar Graham Bell ([source](https://www.countryliving.com/life/g31884612/hard-work-quotes/?slide=15))

Let UwuBot free your mind of having to remember things you need to do! ( T_T) \ (^-^ ) 

## Why UwuBot? ♥️ 
> hellu! 
i am oo woo <: 
how can i be of service today?

 It is, 
- **_EASY_** to learn 😮 
- ~~FAST~~ **_SUPER FAST_** to use 😨 
- And it is **_FREE_**! 😱 

すごいですよ！☆〜（ゝ。∂）

## How to use UwuBot? ♥️ 
All you need to do is,
1. download it from [here](https://github.com/elizabethhky/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you.

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

## What features are there? ♥️ 
- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

elizabethhky committed:
Add UwuBot.java and UwuChat.java

* UwuBot is the main class to be executed.
* UwuChat is the class containing methods that are printed.

elizabethhky committed:
Refactor UwuBot.java and UwuChat.java

* In UwuBot, create UwuChat chat variable so that UwuChat methods no need to be static.
* In UwuChat, refactor userCommands to include "list" command, create helper functions - chatify and listify.

elizabethhky committed:
Add Task.java, Refactor UwuBot and UwuChat

* Create Task class for task objects
* Refactor UwuBot and UwuChat to handle mark and unmark commands

elizabethhky committed:
Add ToDos, Deadline, Event classes and Refactor UwuBot, UwuChat, and Task

* Create new ToDos, Deadline, Event classes for different Task types
* Update UwuBot and UwuChat to accommodate the new task types

elizabethhky committed:
Delete Duke, Run Text-UITesting

* Edit UwuBot, UwuChat to not have Japanese characters
* Refactor runtest.sh to run test cases in input.txt and EXPECTED.TXT

elizabethhky committed:
Add EmptyInputException, IncorrectFormatException, UnknownCommandException, UwuException classes and Refactor UwuBot and UwuChat

elizabethhky committed:
Add NullTaskException class and Refactor UwuBot, UwuChat, and Task class

elizabethhky committed:
Add TaskList and Storage Class, Refactor Events

* Add TaskList and Storage classes to handle and store user's todo list
* Refactor Events to have toStorageString method

elizabethhky committed:
Add UwuDateTime, Add InvalidDateException

* Create UwuDateTime class to handle conversion from string to LocalDateTime object
* Create InvalidDateException to handle invalid dates entered

elizabethhky committed:
Refactor Storage, Create UwuHandler

* Refactor Storage class to load tasks saved from taskList.txt
* Create UwuHandler to separate from UwuBot

elizabethhky committed:
Merge branch 'branch-Level-8'

# Resolved Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/UwuBot.java
#	src/main/java/UwuChat.java

elizabethhky committed:
Add Command

* Create Command as abstract class
* Create different command classes to extend Command
* Refactor UwuHandler to be Parser, UwuChat to be Ui

elizabethhky committed:
Refactor classes into Packages

* Organise classes into uwu package
* Organise further into command, exception and task packages

elizabethhky committed:
Merge branch 'nus-cs2103-AY2223S1:add-gradle-support' into add-gradle-support

elizabethhky committed:
Add JUnit Tests

* Created JUnit Tests for TaskList and ToDos

elizabethhky committed:
Edit mainClassName and jar file name in ShadowJar

elizabethhky committed:
Refactor Parser to throw EmptyInputException

elizabethhky committed:
Add JavaDocs

elizabethhky committed:
Edit code to match coding standard

elizabethhky committed:
Add FindCommand

* Refactor Parser and Ui to accommodate FindCommand

elizabethhky committed:
Merge branch 'branch-A-CodingStandard' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/uwu/Ui.java
#	src/main/java/uwu/command/DeleteCommand.java
#	src/main/java/uwu/command/MarkCommand.java
#	src/main/java/uwu/exception/NullTaskException.java

elizabethhky committed:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/uwu/Parser.java

elizabethhky committed:
Edit FindCommand to match coding standard

fungusta commented on src/main/java/uwu/Parser.java:
```diff
@@ -0,0 +1,52 @@
+                return new DeleteCommand(userCommand);
+            } else if (userCommand.startsWith("find")) {
+                return new FindCommand(userCommand);
+            }else {
```
Should'nt there be a white space before the else statement?

elizabethhky replied:
Thank you for the review! 😊

fungusta commented on src/main/java/uwu/Storage.java:
```diff
@@ -0,0 +1,97 @@
+
+        return result;
+    }
+}
```
I like the code style here 😄

fungusta commented on src/main/java/uwu/Ui.java:
```diff
@@ -0,0 +1,123 @@
+import uwu.task.Task;
+import uwu.task.TaskList;
+
+import java.util.Scanner;
```
I noticed that in Storage.java import statements have different ordering (ie java.* is before uwu.*) Maybe you would like to standardize the ordering

fungusta commented on src/main/java/uwu/Ui.java:
```diff
@@ -0,0 +1,123 @@
+    }
+
+    /**
+     * Leaves chat.
```
Noticed that for the Java Docs of this function in Ui.java are somewhat ambiguous. Does this function end the program? or does it output a leaving message?

fungusta commented on src/main/java/uwu/command/AddCommand.java:
```diff
@@ -0,0 +1,109 @@
+                storage.save(tasks.taskListToStorageString());
+                ui.addTask(todo, tasks.size());
+                break;
+        case "deadline":
```
In the case whenever there is a case statement without a break statement, I believe that there should be an explicit //Fallthrough comment that should be included

fungusta commented on src/main/java/uwu/command/DeleteCommand.java:
```diff
@@ -0,0 +1,61 @@
+ */
+public class DeleteCommand extends Command {
+    /** The index of the task to be deleted. */
+    int index;
```
I noticed that in some classes, such as Uwu bot, access modifiers were indicated for variables, maybe you might want to consider standardizing that for all the classes?

fungusta commented on src/main/java/uwu/command/DeleteCommand.java:
```diff
@@ -0,0 +1,61 @@
+        Task task = tasks.remove(index);
+        storage.save(tasks.taskListToStorageString());
+        ui.deleteTask(task, tasks.size());
+    };
```
Noticed that for some functions, there were semi colons after the curly braces, was wondering why you added them in, care to elaborate?

fungusta reviewed (commented):
Very well written!! 👍 Just some minor tweaks are needed

elizabethhky committed:
Edit minor coding standard violations

jasonqiu212 commented on src/main/java/uwu/Parser.java:
```diff
@@ -0,0 +1,52 @@
+     *                      If the task description is empty.
+     */
+    public static Command parse(String userCommand) throws UwuException {
+            if (userCommand.equals("bye")) {
```
is the indentation here 8 spaces?

elizabethhky replied:
Thank you for the review! 😊

jasonqiu212 commented on src/main/java/uwu/Ui.java:
```diff
@@ -0,0 +1,123 @@
+     * Prints greetings to users.
+     */
+    public void greetUsers() {
+        String greetings = "\thellu!\n\ti am oo woo <:\n\thow can i be of service today?";
```
cute!

jasonqiu212 commented on src/main/java/uwu/Ui.java:
```diff
@@ -0,0 +1,123 @@
+     * Prints a horizontal line to separate chat messages.
+     */
+    public void showLine() {
+        String horizontalLine ="----------------------------------------------------------------------";
```
dont forget a space after `=`!

jasonqiu212 commented on src/main/java/uwu/Ui.java:
```diff
@@ -0,0 +1,123 @@
+            System.out.println("\thm...looks like there are no tasks matching the keyword ><\n\ttry another one~");
+        } else {
+            System.out.println("\there are the tasks i found that match your keyword~ <:" + tasks.taskListToString());
+        }
```
for the methods in this class, I felt that the naming seems to communicate to me that these methods are actually doing what it's saying (eg. `findTask` tells me that this method is for finding a task, rather than displaying the UI that goes with it) I would suggest adding something like `displayFindTask` or `showDeleteTaskMessage`? what do you think?

jasonqiu212 commented on src/main/java/uwu/task/TaskList.java:
```diff
@@ -0,0 +1,101 @@
+/** This class encapsulates a TaskList object. */
+public class TaskList {
+    /** The list of user's tasks. */
+    public ArrayList<Task> userToDoList;
```
would it better to use the plural form for the naming, since it's a collection of items?

jasonqiu212 reviewed (commented):
helloo! your code looks great to me! I really like the separation of commands into individual classes and the cute theme. just some minor code standard and naming issues here and there

elizabethhky committed:
Edit files to pass checkstyle

elizabethhky committed:
Add GUI for UwuBot

elizabethhky committed:
Merge branch 'branch-A-CheckStyle' into branch-Level-10

# Conflicts:
#	src/main/java/uwu/Ui.java
#	src/main/java/uwu/UwuBot.java
#	src/main/java/uwu/command/AddCommand.java
#	src/main/java/uwu/command/Command.java
#	src/main/java/uwu/command/DeleteCommand.java
#	src/main/java/uwu/command/ExitCommand.java
#	src/main/java/uwu/command/FindCommand.java
#	src/main/java/uwu/command/ListCommand.java
#	src/main/java/uwu/command/MarkCommand.java
#	src/main/java/uwu/uwu/Parser.java

elizabethhky committed:
Edit Ui method names

* To make the Ui method names sound clearer, precede Ui method names with "display"

elizabethhky committed:
Add assertions

There are no checks done to prevent programmer's mistakes. These include
FXML components being improperly configured and the storage file not
existing.

Add assertions to check for these mistakes and print out an error
message.

Using assertions is preferable over exceptions as assertions indicate
that the programmer has made a mistake in their code rather than a
mistake made by the user.

elizabethhky committed:
Improve code quality

Although the code worked fine, the code could have been more readable.
There were also some assumptions made about the user's input such as
the user's commands containing spaces. These assumptions can be
dangerous as runtime errors can occur.

Add more classes for commands. Add more exceptions to handle more user
inputs.

elizabethhky committed:
Merge pull request #2 from elizabethhky/branch-A-Assertions

Add assertions

elizabethhky committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/uwu/uwu/Storage.java

elizabethhky committed:
Merge pull request #3 from elizabethhky/branch-A-CodeQuality

Improve code quality

elizabethhky committed:
Add HelpCommand

elizabethhky committed:
Add CI

elizabethhky committed:
Edit classes to follow coding standard

Add Javadocs to HelpCommand and UI

elizabethhky committed:
Refactor Storage load()

* Abstract out decodeTask method from load method to avoid arrow-head
code
* Edit welcome message to include help command

elizabethhky committed:
Improve GUI

* Add PromptText in TextField
* Add listener to adjust scrollPane based on user's scroll
* Make window resizable

elizabethhky committed:
Add a User Guide

elizabethhky committed:
Edit DialogBox to fit coding standard

elizabethhky committed:
Improve GUI

* Disable Send button if user does not type in TextField

elizabethhky committed:
Fix bugs in MarkCommand and UnmarkCommand

* Previously did not show error messages for "mark 0" and "unmark 0"

elizabethhky committed:
Add time delay to exit program

* Add time delay of 1 second after ExitCommand to exit program

elizabethhky committed:
Update README.md
